packages/vscode-extension/src/devices/preview.ts

```diff
@@ -102,23 +116,23 @@ export class Preview implements Disposable {
     this.subprocess?.stdin?.write("pointer show false\n");
   }
 
-  public startReplays() {
+  public startVideoRecording(videoId: string) {
```

Comment:
It is a wrong level of abstraction here. We assume in the callback that IDs can be "recording" and "replay" so we shouldn't allow any string as videoID, as if anything else is passed, the callback would never trigger.

One way would be to specify allowed values in typescript, but I think that this method shouldn't expose videoID as a parameter and just make it an implementation detail.

Reply:
I've made the changes. Let me know if they look as you expected. Additionally, I've added 'sendCommandOrThrow' to reduce code duplication.